packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/TextBoundsProvider.java

```diff
@@ -72,23 +88,86 @@ public TextBounds computeBounds(LabelStyle labelStyle, String text) {
             }
         }
 
+        boolean isIcon = !labelStyle.getIconURL().isEmpty();
+        Position alignment = this.getAlignment(labelBounds, isIcon, true);
+        Size size = this.getSize(labelBounds, isIcon);
+
+        return new TextBounds(size, alignment);
+    }
+
+    public TextBounds computeAutoWrapBounds(LabelStyle labelStyle, String text, double maxWidth) {
+        Rectangle2D labelBounds = new Rectangle();
+        if (text == null || text.isEmpty()) {
+            labelBounds.setFrame(0, 0, 0, 0);
+        } else {
+            Font font = this.getFont(labelStyle).deriveFont(AFFINE_TRANSFORM);
```

Comment:
I assume the `.deriveFont(AFFINE_TRANSFORM)` is to make sure the size computed in the backend match what's displayed on the front. Shouldn't this also be called in the plain `computeBounds`?

Reply:
I don't want to modify `computeBounds`. It is out of scope.